pkg/network/tracer/tracer.go

```diff
@@ -182,6 +185,19 @@ func newTracer(cfg *config.Config, telemetryComponent telemetryComponent.Compone
 	tr.reverseDNS = newReverseDNS(cfg, telemetryComponent)
 	tr.usmMonitor = newUSMMonitor(cfg, tr.ebpfTracer)
 
+	// Setup the connection_protocol map cleaner if protocol classification is enabled and USM is disabled
+	if cfg.ProtocolClassificationEnabled && !usmconfig.IsUSMSupportedAndEnabled(cfg) {
```

Comment:
Should this be `cfg.ProtocolClassificationEnabled || usmconfig.IsUSMSupportedAndEnabled(cfg)` and then we don't use the map in the usm monitor/probe?

Reply:
About initializing it only in npm, it might cause issues in our UTs, as we don't run the tracer, but only USM

Reply:
let me try to fix the UTs

Reply:
update: this didn't actually break any UTs

Reply:
it can create flaky tests as the map leaks entries

Reply:
if you run full system-probe the code is fine as is today
but, the tests are prune to errors, as we run only usm in the tests (without initializing network_tracer, and hence, without the map cleaner).

Reply:
Reviewed again - the code itself works as expected.
As for tests, it is unlikely to help (or cause any issue), as the map cleaner runs every 3m, and deletes connections longer than 5m. There is no test today that lasts long enough to be affected by the cleaner.